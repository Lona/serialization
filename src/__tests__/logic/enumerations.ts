import {
  parse as parseSwift,
  print as printSwift,
} from '../../convert/swift/logic'

jest.mock('uuid', () => ({ v4: () => '0' }))

it('converts enumerations', () => {
  const inputSource = `enum Foo {
  case bar(Number, String)
  case baz(label1: Number, label2: String)
}`

  const json = parseSwift(inputSource, {
    startRule: 'program',
  })

  expect(json).toMatchSnapshot()

  const source = printSwift(json)

  expect(source).toBe(inputSource)
})
