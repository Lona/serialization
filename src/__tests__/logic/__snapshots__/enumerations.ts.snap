// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Enumerations converts code -> json 1`] = `
Object {
  "data": Object {
    "block": Array [
      Object {
        "data": Object {
          "content": Object {
            "data": Object {
              "cases": Array [
                Object {
                  "data": Object {
                    "associatedValues": Array [
                      Object {
                        "data": Object {
                          "annotation": Object {
                            "data": Object {
                              "genericArguments": Array [],
                              "id": "0",
                              "identifier": Object {
                                "id": "0",
                                "isPlaceholder": false,
                                "string": "Number",
                              },
                            },
                            "type": "typeIdentifier",
                          },
                          "id": "0",
                        },
                        "type": "associatedValue",
                      },
                      Object {
                        "data": Object {
                          "annotation": Object {
                            "data": Object {
                              "genericArguments": Array [],
                              "id": "0",
                              "identifier": Object {
                                "id": "0",
                                "isPlaceholder": false,
                                "string": "String",
                              },
                            },
                            "type": "typeIdentifier",
                          },
                          "id": "0",
                        },
                        "type": "associatedValue",
                      },
                      Object {
                        "data": Object {
                          "id": "0",
                        },
                        "type": "placeholder",
                      },
                    ],
                    "id": "0",
                    "name": Object {
                      "id": "0",
                      "name": "bar",
                    },
                  },
                  "type": "enumerationCase",
                },
                Object {
                  "data": Object {
                    "associatedValues": Array [
                      Object {
                        "data": Object {
                          "annotation": Object {
                            "data": Object {
                              "genericArguments": Array [],
                              "id": "0",
                              "identifier": Object {
                                "id": "0",
                                "isPlaceholder": false,
                                "string": "Number",
                              },
                            },
                            "type": "typeIdentifier",
                          },
                          "id": "0",
                          "label": Object {
                            "id": "0",
                            "name": "label1",
                          },
                        },
                        "type": "associatedValue",
                      },
                      Object {
                        "data": Object {
                          "annotation": Object {
                            "data": Object {
                              "genericArguments": Array [],
                              "id": "0",
                              "identifier": Object {
                                "id": "0",
                                "isPlaceholder": false,
                                "string": "String",
                              },
                            },
                            "type": "typeIdentifier",
                          },
                          "id": "0",
                          "label": Object {
                            "id": "0",
                            "name": "label2",
                          },
                        },
                        "type": "associatedValue",
                      },
                      Object {
                        "data": Object {
                          "id": "0",
                        },
                        "type": "placeholder",
                      },
                    ],
                    "id": "0",
                    "name": Object {
                      "id": "0",
                      "name": "baz",
                    },
                  },
                  "type": "enumerationCase",
                },
                Object {
                  "data": Object {
                    "id": "0",
                  },
                  "type": "placeholder",
                },
              ],
              "genericParameters": Array [
                Object {
                  "data": Object {
                    "id": "0",
                  },
                  "type": "placeholder",
                },
              ],
              "id": "0",
              "name": Object {
                "id": "0",
                "name": "Foo",
              },
            },
            "type": "enumeration",
          },
          "id": "0",
        },
        "type": "declaration",
      },
      Object {
        "data": Object {
          "id": "0",
        },
        "type": "placeholder",
      },
    ],
    "id": "0",
  },
  "type": "program",
}
`;

exports[`converts enumerations 1`] = `
Object {
  "data": Object {
    "block": Array [
      Object {
        "data": Object {
          "content": Object {
            "data": Object {
              "cases": Array [
                Object {
                  "data": Object {
                    "associatedValues": Array [
                      Object {
                        "data": Object {
                          "annotation": Object {
                            "data": Object {
                              "genericArguments": Array [],
                              "id": "0",
                              "identifier": Object {
                                "id": "0",
                                "isPlaceholder": false,
                                "string": "Number",
                              },
                            },
                            "type": "typeIdentifier",
                          },
                          "id": "0",
                        },
                        "type": "associatedValue",
                      },
                      Object {
                        "data": Object {
                          "annotation": Object {
                            "data": Object {
                              "genericArguments": Array [],
                              "id": "0",
                              "identifier": Object {
                                "id": "0",
                                "isPlaceholder": false,
                                "string": "String",
                              },
                            },
                            "type": "typeIdentifier",
                          },
                          "id": "0",
                        },
                        "type": "associatedValue",
                      },
                      Object {
                        "data": Object {
                          "id": "0",
                        },
                        "type": "placeholder",
                      },
                    ],
                    "id": "0",
                    "name": Object {
                      "id": "0",
                      "name": "bar",
                    },
                  },
                  "type": "enumerationCase",
                },
                Object {
                  "data": Object {
                    "associatedValues": Array [
                      Object {
                        "data": Object {
                          "annotation": Object {
                            "data": Object {
                              "genericArguments": Array [],
                              "id": "0",
                              "identifier": Object {
                                "id": "0",
                                "isPlaceholder": false,
                                "string": "Number",
                              },
                            },
                            "type": "typeIdentifier",
                          },
                          "id": "0",
                          "label": Object {
                            "id": "0",
                            "name": "label1",
                          },
                        },
                        "type": "associatedValue",
                      },
                      Object {
                        "data": Object {
                          "annotation": Object {
                            "data": Object {
                              "genericArguments": Array [],
                              "id": "0",
                              "identifier": Object {
                                "id": "0",
                                "isPlaceholder": false,
                                "string": "String",
                              },
                            },
                            "type": "typeIdentifier",
                          },
                          "id": "0",
                          "label": Object {
                            "id": "0",
                            "name": "label2",
                          },
                        },
                        "type": "associatedValue",
                      },
                      Object {
                        "data": Object {
                          "id": "0",
                        },
                        "type": "placeholder",
                      },
                    ],
                    "id": "0",
                    "name": Object {
                      "id": "0",
                      "name": "baz",
                    },
                  },
                  "type": "enumerationCase",
                },
                Object {
                  "data": Object {
                    "id": "0",
                  },
                  "type": "placeholder",
                },
              ],
              "genericParameters": Array [
                Object {
                  "data": Object {
                    "id": "0",
                  },
                  "type": "placeholder",
                },
              ],
              "id": "0",
              "name": Object {
                "id": "0",
                "name": "Foo",
              },
            },
            "type": "enumeration",
          },
          "id": "0",
        },
        "type": "declaration",
      },
      Object {
        "data": Object {
          "id": "0",
        },
        "type": "placeholder",
      },
    ],
    "id": "0",
  },
  "type": "program",
}
`;
